common.yaml.jinja

```diff
@@ -54,7 +54,7 @@ services:
 
   smtpreal:
     image: ghcr.io/docker-mailserver/docker-mailserver:{{ smtp_relay_version }}
-    domainname: "{{ smtp_canonical_default }}"
+    hostname: "smtp.{{ smtp_canonical_default }}"
```

Comment:
From what I gather from the documentation, hostname must be used: https://docker-mailserver.github.io/docker-mailserver/latest/examples/tutorials/basic-installation/

But yes, it is not necessary to hardcode the service. It should not really affect nothing since the mail service is only used for outgoing mail... and odoo uses the network alias created in 'prod.yaml'.

I have also updated the rest of the hostnames. No one should be affected by these changes.